omnixai/explanations/ranking/agnostic/validity.py

```diff
@@ -0,0 +1,200 @@
+#
+# Copyright (c) 2022 salesforce.com, inc.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+#
+"""
+Counterfactual explanations.
+"""
+import numpy as np
+from ...base import ExplanationBase, DashFigure
+
+
+class ValidExplanation(ExplanationBase):
+    """
+    The class for ranking explanation results.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.explanations = []
+
+    def __repr__(self):
+        return repr(self.explanations)
+
+    def set(self, query, df, top_features, validity, **kwargs):
```

Comment:
To make it consistent with other Explanation classes, we can change the function name "set" to "add".

Reply:
add is for multiple explanations. but here it is just a single explanation.